application/client/src/app/ui/views/toolbar/chart/zoomer/state.ts

```diff
@@ -0,0 +1,230 @@
+import { ISearchMap } from '@platform/interfaces/interface.rust.api.general';
+import { Instance as Logger } from '@platform/env/logger';
+import { Session } from '@service/session';
+import { Chart, ChartDataset } from 'chart.js';
+import { StoredEntity } from '@service/session/dependencies/search/store';
+import { FilterRequest } from '@service/session/dependencies/search/filters/request';
+import { ViewContainerRef } from '@angular/core';
+import { IlcInterface } from '@service/ilc';
+import { ChangesDetector } from '@ui/env/extentions/changes';
+import { getChartService, Service } from '../service';
+
+export class State {
+    public isCursorVisible: boolean = true;
+
+    private _parent!: IlcInterface & ChangesDetector;
+    private _filters!: Chart;
+    private _session!: Session;
+    private _service!: Service;
+    private _vcRef!: ViewContainerRef;
+    private _logger!: Logger;
+    private _labelCount: number = 0;
+    private _datasets: ChartDataset<'bar', number[]>[] = [];
+    private _map: ISearchMap = [];
+    private _loading: boolean = false;
+    private _width: number = 0;
+
+    public bind(
+        parent: IlcInterface & ChangesDetector,
+        session: Session,
+        vcRef: ViewContainerRef,
+        logger: Logger,
+    ) {
```

Comment:
BTW, as soon as you pass `parent` as `IlcInterface`, you don't need `logger: Logger,`, because you can write logs directly `parent.log().error('bla-bla')`

Reply:
one more field out - less complexity, less bugs